soh/soh/Enhancements/randomizer/randomizer_check_objects.cpp

```diff
@@ -894,11 +894,34 @@ std::map<RandomizerCheck, RandomizerCheckObject> RandomizerCheckObjects::GetAllR
     return rcObjects;
 }
 
+std::map<SceneID, RandomizerCheckArea> rcAreaBySceneID = {};
+std::map<SceneID, RandomizerCheckArea> RandomizerCheckObjects::GetAllRCAreaBySceneID() {
+    //memoize on first request
+    if (rcAreaBySceneID.size() == 0) {
+        for (auto& [randomizerCheck, rcObject] : rcObjects) {
+            rcAreaBySceneID[rcObject.sceneId] = rcObject.rcArea;
+        }
+        //Add checkless Hyrule Market areas to the area return
+        for (int id = (int)SCENE_ENTRA; id <= (int)SCENE_MARKET_RUINS; id++)
+            rcAreaBySceneID[(SceneID)id] = RCAREA_MARKET;
+    }
+    return rcAreaBySceneID;
+}
+
+RandomizerCheckArea RandomizerCheckObjects::GetRCAreaBySceneID(SceneID sceneId) {
+    std::map<SceneID, RandomizerCheckArea> areas = GetAllRCAreaBySceneID();
+    auto areaIt = areas.find(sceneId);
+    if (areaIt == areas.end())
+        return RCAREA_INVALID;
+    else
+        return areaIt->second;
+}
+
 void RandomizerCheckObjects::UpdateImGuiVisibility() {
     for (auto& [randomizerCheck, locationIt] : rcObjects) {
         locationIt.visibleInImgui = (
             (locationIt.vOrMQ != RCVORMQ_MQ) && // don't show MQ checks until we support MQ
-            (locationIt.rcType != RCTYPE_SHOP) && // don't show shop items until we have shopsanity
+            ((locationIt.rcType != RCTYPE_SHOP) || CVar_GetS32("gRandomizeShopsanity", 0) > 1) && // 1 is the value for "0 random items" for shop
```

Comment:
```suggestion
            ((locationIt.rcType != RCTYPE_SHOP) || (
                (CVar_GetS32("gRandomizeShopsanity", RO_SHOPSANITY_OFF) != RO_SHOPSANITY_OFF) &&
                (CVar_GetS32("gRandomizeShopsanity", RO_SHOPSANITY_OFF) != RO_SHOPSANITY_ZERO_ITEMS)
            )) &&
```

Reply:
In reality, I probably should be removing code changes to this method entirely from this code, and only apply them via #2070.

Reply:
Resolved with merge (with expected conflicts from #2070) ee09aa07ee81ad2c1fe48c752c3c971442272f8d